### DISCUSSION TITLE
How to remove the list number of catalogue (TOC)

### DISCUSSION THREAD

**Discussion by panxiuwen** [hexojs/hexo#5172] [Question / Answer]
It is repeat with what I writen, and I don't like it, so how to remove the list number of catalogue
![QQ截图20230309002622](https://user-images.githubusercontent.com/35104739/223773794-312cf91e-5e5f-4aa0-87e3-eaa1907e1106.png)

**Comment by lorezyra** [comment]
Which theme are you running? Are you using the built-in [`toc_obj` from `hexo-util`](https://github.com/hexojs/hexo-util/blob/master/lib/toc_obj.ts) node_module?

If you are using the built-in, then you can use the (html element) attribute [`data-toc-unnumbered`](https://github.com/hexojs/hexo-util/blob/d31764c4058ff71802490c265b9515b4dd8efc7a/lib/toc_obj.ts#L21) on your TOC `<div data-toc-unnumbered>`.

**Reply by panxiuwen** [reply]
Dear @lorezyra ,thanks so much for your reply.
I did use the built-in toc_obj, with the theme of [Tranquilpeak](https://github.com/LouisBarranqueiro/hexo-theme-tranquilpeak), just used ```<!-- toc -->``` to generate the catalogue.
Despite I got the help of you, refer the [Hexo_doc](https://hexo.io/docs/helpers.html#toc) and [this](https://github.com/hexojs/hexo/pull/4871), but I still don't know how to use a CSS or modify toc_obj.js to custom what I want.
I'm not so proficient to deal with it, could your please give me a example for me? Thank you
Remove all the list number is better

**Reply by lorezyra** [reply]
I think you'll find your clue here: https://github.com/hexojs/hexo-util/pull/269#issuecomment-1019234041

Can you confirm which renderer you are using?

**Reply by lorezyra** [reply]
BTW, I built my own theme and new ToC too. Sometimes you can't get everything you want in a theme. So, you may need to build something new for your use-case.

**Reply by panxiuwen** [reply]
Dear @lorezyra ,thanks so much for your reply.
I used hexo-renderer-marked, and try the markdown-it-attrs to add the data-toc-unnumbered.
But it still don't work, it render as "XXX  { data-toc-unnumbered="true" } " instead of ```<h1 class="data-toc-unnumbered="true"">XXX</h1>```, I don't know what's wrong with it.
In addition, I want to delete all the list number, so it is inconvenient to add the style to all the heads.
Maybe it's a good idea to modify some code, I try to test the toc_obj.js, but fail to do this

**Reply by lorezyra** [reply]
I will add this use-case to my own ToC feature list. When done, perhaps I can share. (It may be a while as I have a lot of issues to complete.) You can find my design here: https://2022.Blog.RichieBartlett.com .

**Reply by panxiuwen** [reply]
Thanks for your reply, I am waiting for your advice.This is [My websit](https://panxiuwen.github.io/).

**Comment by lorezyra** [comment]
As the default ToC is not to your liking, I recommend replacing it with a plugin. Checkout the packages available on [NPMjs.com](https://www.npmjs.com/search?q=hexo%20toc).

- https://www.npmjs.com/package/hexo-toc-customized

**Reply by panxiuwen** [reply]
Thanks your very much @lorezyra, I will try it later, after that ,we can discuss it.
